src/tests/intg/test_ssh_pubkey.py

```diff
@@ -1,341 +0,0 @@
-    conf.close()
-    os.chmod(config.CONF_PATH, stat.S_IRUSR | stat.S_IWUSR)
-
-
-def cleanup_conf_file():
-    """Remove sssd.conf, if it exists"""
-    if os.path.lexists(config.CONF_PATH):
-        os.unlink(config.CONF_PATH)
-
-
-def create_conf_cleanup(request):
-    """Add teardown for removing sssd.conf"""
-    request.addfinalizer(cleanup_conf_file)
-
-
-def create_conf_fixture(request, contents):
-    """
-    Create sssd.conf with specified contents and add teardown for removing it
-    """
-    create_conf_file(contents)
-    create_conf_cleanup(request)
-
-
-def create_sssd_process():
-    """Start the SSSD process"""
-    if subprocess.call(["sssd", "-D", "--logger=files"]) != 0:
-        raise Exception("sssd start failed")
-
-
-def get_sssd_pid():
-    pid_file = open(config.PIDFILE_PATH, "r")
-    pid = int(pid_file.read())
-    return pid
-
-
-def cleanup_sssd_process():
-    """Stop the SSSD process and remove its state"""
-    try:
-        pid = get_sssd_pid()
-        os.kill(pid, signal.SIGTERM)
-        while True:
-            try:
-                os.kill(pid, signal.SIGCONT)
-            except OSError:
-                break
-            time.sleep(1)
-    except OSError:
-        pass
-    for path in os.listdir(config.DB_PATH):
-        os.unlink(config.DB_PATH + "/" + path)
-    for path in os.listdir(config.MCACHE_PATH):
-        os.unlink(config.MCACHE_PATH + "/" + path)
-
-
-def create_sssd_fixture(request):
-    """Start SSSD and add teardown for stopping it and removing its state"""
-    create_sssd_process()
-    create_sssd_cleanup(request)
-
-
-def create_sssd_cleanup(request):
-    """Add teardown for stopping SSSD and removing its state"""
-    request.addfinalizer(cleanup_sssd_process)
-
-
-@pytest.fixture
-def add_user_with_ssh_key(request, ldap_conn):
-    ent_list = ldap_ent.List(ldap_conn.ds_inst.base_dn)
-    ent_list.add_user("user1", 1001, 2001,
-                      sshPubKey=(USER1_PUBKEY1, USER1_PUBKEY2))
-    ent_list.add_user("user2", 1002, 2001)
-    create_ldap_fixture(request, ldap_conn, ent_list)
-
-    config.PAM_CERT_DB_PATH = os.environ['PAM_CERT_DB_PATH']
-    conf = format_basic_conf(ldap_conn, SCHEMA_RFC2307_BIS, config)
-    create_conf_fixture(request, conf)
-    create_sssd_fixture(request)
-    return None
-
-
-def test_ssh_pubkey_retrieve(add_user_with_ssh_key):
-    """
-    Test that we can retrieve an SSH public key for a user who has one
-    and can't retrieve a key for a user who does not have one.
-    """
-    sshpubkey = get_call_output(["sss_ssh_authorizedkeys", "user1"])
-    assert sshpubkey == USER1_PUBKEY1 + '\n' + USER1_PUBKEY2 + '\n'
-
-    sshpubkey = get_call_output(["sss_ssh_authorizedkeys", "user2"])
-    assert len(sshpubkey) == 0
-
-
-def test_ssh_pubkey_retrieve_cert(add_user_with_ssh_cert):
-    """
-    Test that we can retrieve an SSH public key derived from a cert in ldap.
-    Compare with the sshpubkey derived via ssh-keygen, they should match.
```

Comment:
Looks like we are removing test coverage for public key derived from certificate, is it okay? `        ldap_user_certificate = userCertificate;binary` in format_basic_conf method above.

Reply:
I think it's okay, because most users would use ssh-keygen, but I maybe overlooking something because I'm not all that familiar with smartcards. Is it a common scenario to take a certificate from a smartcard and convert it to an SSH key? Is it even important? Maybe it's already covered, @spoore1 ?

Reply:
The certificate has to be converted into a public key for use with SSH.   We have some coverage downstream but, it's going to be a while still before that can be ported.

Reply:
@spoore1 Can we plan it for a smartcard test, and push this forward?

Reply:
Hi,

`test_ssh_pubkey_retrieve_cert` tests the feature to generate an ssh-key from the public-key of the Smartcart so that the Smartcard can be used for ssh pubkey authentication. Wouldn't it be possible to check this feature in a similar way as the integration test, i.e. use the certificates generated by the SSSD test CA and compare the output of `sss_ssh_authorized_keys` with the corresponding ssh-keys also generated by the SSSD test CA?

bye,
Sumit